src/main/java/io/vertx/serviceproxy/AuthenticationInterceptor.java

```diff
@@ -0,0 +1,19 @@
+package io.vertx.serviceproxy;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.ext.auth.authentication.AuthenticationProvider;
+
+/**
+ * Create an event bus service interceptor that will provide an authentication check
+ */
+@VertxGen
+public interface AuthenticationInterceptor extends ServiceInterceptor {
+
```

Comment:
For completeness we should have a factory method here to avoid exposing the `impl` package:

```java
static AuthenticationInterceptor create() {
  return new AuthenticationInterceptorImpl();
}
```

This hiddes the impl package and class.